crates/flux-middle/src/queries.rs

```diff
@@ -77,6 +77,11 @@ pub enum QueryErr {
         trait_id: DefId,
         name: Symbol,
     },
+    IncompatibleSubtyping {
+        msg: String,
+        span: Span,
+        super_span: Option<Span>,
+    },
```

Comment:
This shouldn't be here. `QueryErr` is for errors that happen deep inside a query but need to be reported somewhere else. If anything, this should be an `InferErr`, but see comment below.